File: components/redis-handling-session/src/main/java/com/sv/youapp/component/authorization/converter/BytesToUsernamePasswordAuthenticationTokenConverter.java
package com.sv.youapp.component.authorization.converter;

import org.springframework.core.convert.converter.Converter;
import org.springframework.data.convert.ReadingConverter;
import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;

@ReadingConverter
public class BytesToUsernamePasswordAuthenticationTokenConverter
implements Converter<byte[], UsernamePasswordAuthenticationToken> {

    private final Jackson2JsonRedisSerializer<UsernamePasswordAuthenticationToken> serializer;

    public BytesToUsernamePasswordAuthenticationTokenConverter(
            Jackson2JsonRedisSerializer<UsernamePasswordAuthenticationToken> serializer
            ) {
        this.serializer = serializer;
    }

    @Override
    public UsernamePasswordAuthenticationToken convert(byte[] value) {
        return this.serializer.deserialize(value);
    }
}

File: platform-instances/authorization-server/src/main/kotlin/com/sv/youapp/service/authorization/converter/BytesToUsernamePasswordAuthenticationTokenConverter.kt
package com.sv.youapp.service.authorization.converter

import org.springframework.core.convert.converter.Converter
import org.springframework.data.convert.ReadingConverter
import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken

@ReadingConverter
class BytesToUsernamePasswordAuthenticationTokenConverter(val serializer: Jackson2JsonRedisSerializer<UsernamePasswordAuthenticationToken?>) :
    Converter<ByteArray?, UsernamePasswordAuthenticationToken?> {
    override fun convert(value: ByteArray): UsernamePasswordAuthenticationToken? {
        return this.serializer.deserialize(value)
    }
}
